user/echo.c

```diff
@@ -1,19 +1,56 @@
 #include "kernel/types.h"
 #include "kernel/stat.h"
+#include <stddef.h>
 #include "user/user.h"
 
 int
 main(int argc, char *argv[])
 {
   int i;
-
-  for(i = 1; i < argc; i++){
-    write(1, argv[i], strlen(argv[i]));
-    if(i + 1 < argc){
-      write(1, " ", 1);
-    } else {
-      write(1, "\n", 1);
+  void echo_function_enhanced(int argc, char *argv[], int start_val){
```

Comment:
Interesting to see nested function in C seem to work. https://stackoverflow.com/questions/2608158/nested-function-in-c. Could we move this outside of main. Might cause portability issues if some one is else if using different compiler.